db/hints/internal/hint_storage.cc

```diff
@@ -168,71 +168,82 @@ void rebalance_segments_for(const sstring& ep, size_t segments_per_shard,
+                current_segments_to_move.splice(current_segments_to_move.end(), shard_segments,
+                        std::next(shard_segments.begin(), q), shard_segments.end());
             }
         }
     }
 
-    // Since N (a total number of segments to a specific destination) may be not a multiple of S (a current number of
-    // shards) we will distribute files in two passes:
+    // Since N (a total number of segments to a specific destination) may be not a multiple
+    // of S (the current number of shards) we will distribute files in two passes:
     //    * if N = S * q + r, then
     //       * one pass for segments_per_shard = q
     //       * another one for segments_per_shard = q + 1.
     //
     // This way we will ensure as close to the perfect distribution as possible.
     //
-    // Right till this point we haven't moved any segments. However we have created a logical separation of segments
-    // into two groups:
+    // Right till this point we haven't moved any segments. However we have created a logical
+    // separation of segments into two groups:
     //    * Segments that are not going to be moved: segments in the segments_map.
     //    * Segments that are going to be moved: segments in the segments_to_move.
     //
-    // rebalance_segments_for() is going to consume segments from segments_to_move and move them to corresponding
-    // lists in the segments_map AND actually move segments to the corresponding shard's sub-directory till the requested
-    // segments_per_shard level is reached (see more details in the description of rebalance_segments_for()).
-    for (auto& [ep, N] : per_ep_hints) {
-        size_t q = N / smp::count;
-        size_t r = N - q * smp::count;
+    // rebalance_segments_for() is going to consume segments from segments_to_move
+    // and move them to corresponding lists in the segments_map AND actually move segments
+    // to the corresponding shard's sub-directory till the requested segments_per_shard level
+    // is reached (see more details in the description of rebalance_segments_for()).
+    for (const auto& [ep, N] : per_ep_hints) {
+        const size_t q = N / smp::count;
+        const size_t r = N - q * smp::count;
         auto& current_segments_to_move = segments_to_move[ep];
         auto& current_segments_map = segments_map[ep];
 
         if (q) {
-            rebalance_segments_for(ep, q, hints_directory, current_segments_map, current_segments_to_move);
+            rebalance_segments_for(ep, q, hint_directory, current_segments_map,
+                    current_segments_to_move);
         }
 
         if (r) {
-            rebalance_segments_for(ep, q + 1, hints_directory, current_segments_map, current_segments_to_move);
+            rebalance_segments_for(ep, q + 1, hint_directory, current_segments_map,
+                    current_segments_to_move);
```

Comment:
Nit: the seastar coding style requires lines to be at most 120 characters in length. Lines shorter than that limit are fine, so no need to change them - for example, these ones don't seem to need such an adjustment.